reactor-core/src/main/java/reactor/core/publisher/MonoZip.java

```diff
@@ -522,8 +524,8 @@ public void onComplete() {
 			}
 		}
 
-		void cancel() {
-			Operators.terminate(S, this);
+		boolean cancel() {
+			return !Operators.terminate(S, this);
```

Comment:
`Operators.terminate` returns `false` if already cancelled, so the `!` turns that to true. Above, we discard if `cancel` returns true, ie. if the subscription was already cancelled so I think that `!` is a mistake.

Reply:
This is by desing. Here - > https://github.com/reactor/reactor-core/pull/3081/files/12382e0896841a060cae627b6e406521bbb58e6a..c0995ca264f6dd956d35a92c12a70a51ff023fef#diff-888f0ff88ba0f8cb4bfd50bfe62c1c71d8b343fcb0741f0dd809a6bb5c240569R474, I do replace subscription so by doing we indicate that value is set, so when we call cancel, if it is already terminated, it means that a caller can discard value, otherwise, if it is not terminated,then a value (if present) is discarded at onNext